service/history/api/pollupdate/api.go

```diff
@@ -28,21 +28,29 @@ import (
 	"context"
 	"fmt"
 
+	"go.temporal.io/api/enums/v1"
 	"go.temporal.io/api/serviceerror"
 	"go.temporal.io/api/workflowservice/v1"
 
 	"go.temporal.io/server/api/historyservice/v1"
 	"go.temporal.io/server/common/definition"
+	"go.temporal.io/server/common/namespace"
 	"go.temporal.io/server/service/history/api"
+	"go.temporal.io/server/service/history/shard"
 	"go.temporal.io/server/service/history/workflow"
 	"go.temporal.io/server/service/history/workflow/update"
 )
 
 func Invoke(
 	ctx context.Context,
 	req *historyservice.PollWorkflowExecutionUpdateRequest,
+	shardContext shard.Context,
 	ctxLookup api.WorkflowConsistencyChecker,
 ) (*historyservice.PollWorkflowExecutionUpdateResponse, error) {
+	waitStage := req.GetRequest().GetWaitPolicy().GetLifecycleStage()
+	if waitStage != enums.UPDATE_WORKFLOW_EXECUTION_LIFECYCLE_STAGE_COMPLETED {
+		return nil, serviceerror.NewUnimplemented(fmt.Sprintf("support for LifecycleStage=%v is not implemented", waitStage))
```

Comment:
Hmm... I think unimplemented counts against our SLA, maybe better to use invalid argument here.
CC @yiminc

Reply:
Changed to `InvalidArgument`